nixosConfigurations/services/steward.nix

```diff
@@ -2,11 +2,20 @@
   self,
   flake-utils,
   sops-nix,
+  nixpkgs,
   ...
 }:
 with flake-utils.lib.system; let
-  mkSteward = self.lib.hosts.mkService [
+  mkSteward = useNginx: self.lib.hosts.mkService [
     sops-nix.nixosModules.sops
+    ({config, ...}: nixpkgs.lib.mkIf useNginx {
+      services.nginx.virtualHosts.${config.networking.fqdn} = {
+        enableACME = true;
+        forceSSL = true;
+        http2 = false;
+        locations."/".proxyPass = "http://localhost:3000";
+      };
+    })
```

Comment:
This would be a bit more idiomatic and should evaluate quicker

```diff
diff --git a/nixosConfigurations/services/steward.nix b/nixosConfigurations/services/steward.nix
index 1abf008..a61eef7 100644
--- a/nixosConfigurations/services/steward.nix
+++ b/nixosConfigurations/services/steward.nix
@@ -1,46 +1,47 @@
 {
   self,
   flake-utils,
+  nixlib,
   sops-nix,
-  nixpkgs,
   ...
 }:
 with flake-utils.lib.system; let
-  mkSteward = useNginx: self.lib.hosts.mkService [
-    sops-nix.nixosModules.sops
-    ({config, ...}: nixpkgs.lib.mkIf useNginx {
-      services.nginx.virtualHosts.${config.networking.fqdn} = {
-        enableACME = true;
-        forceSSL = true;
-        http2 = false;
-        locations."/".proxyPass = "http://localhost:3000";
-      };
-    })
-    ({
-      config,
-      pkgs,
-      ...
-    }: {
-      services.steward.certFile = pkgs.writeText "steward.crt" (builtins.readFile "${self}/hosts/${config.networking.fqdn}/steward.crt");
-      services.steward.enable = true;
-      services.steward.keyFile = config.sops.secrets.key.path;
+  mkSteward = useNginx:
+    self.lib.hosts.mkService ([
+        sops-nix.nixosModules.sops
+        ({
+          config,
+          pkgs,
+          ...
+        }: {
+          services.steward.certFile = pkgs.writeText "steward.crt" (builtins.readFile "${self}/hosts/${config.networking.fqdn}/steward.crt");
+          services.steward.enable = true;
+          services.steward.keyFile = config.sops.secrets.key.path;
 
-      sops.age.sshKeyPaths = ["/etc/ssh/ssh_host_ed25519_key"];
-      sops.secrets.key.format = "binary";
-      sops.secrets.key.mode = "0000";
-      sops.secrets.key.restartUnits = ["steward.service"];
-      sops.secrets.key.sopsFile = "${self}/hosts/${config.networking.fqdn}/steward.key";
+          sops.age.sshKeyPaths = ["/etc/ssh/ssh_host_ed25519_key"];
+          sops.secrets.key.format = "binary";
+          sops.secrets.key.mode = "0000";
+          sops.secrets.key.restartUnits = ["steward.service"];
+          sops.secrets.key.sopsFile = "${self}/hosts/${config.networking.fqdn}/steward.key";
 
-      systemd.services.steward = self.lib.systemd.withSecret config pkgs "steward" "key";
+          systemd.services.steward = self.lib.systemd.withSecret config pkgs "steward" "key";
 
-      # Workaround for https://github.com/profianinc/infrastructure/issues/109
+          # Workaround for https://github.com/profianinc/infrastructure/issues/109
 
-      users.groups.steward = {};
+          users.groups.steward = {};
 
-      users.users.steward.isSystemUser = true;
-      users.users.steward.group = config.users.groups.steward.name;
-    })
-  ];
+          users.users.steward.isSystemUser = true;
+          users.users.steward.group = config.users.groups.steward.name;
+        })
+      ]
+      ++ nixlib.lib.optional useNginx ({config, ...}: {
+        services.nginx.virtualHosts.${config.networking.fqdn} = {
+          enableACME = true;
+          forceSSL = true;
+          http2 = false;
+          locations."/".proxyPass = "http://localhost:3000";
+        };
+      }));
 
   attest-testing = mkSteward true x86_64-linux [
     ({pkgs, ...}: {
```

Reply:
Thanks for that.
Do you know of a tool that could automatically do or check the formatting?

Reply:
`nix fmt`